### PR TITLE
add Lab1 3variant

### PR DESCRIPTION
**Pull request by maksimkha**

### CODE REVIEW

File: Lab1/Program.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                    SendMessage();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+        }
+        private static void SendMessage()
+        {
+            UdpClient sender = new UdpClient();
```
Comment (gmltA) [inline Lab1/Program.cs:44]: U sure you have to create sender each time you send a message?

File: Lab1/Program.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                    Console.Clear();
+                    foreach (var note in historyList)
+                    {
+                        Console.WriteLine(note.Item1 + " " + note.Item2);
+                    }
+                    id++;
+                    if (historyList.Count > 1)
+                        if (tempId != historyList[historyList.Count - 2].Item1 + 1)
+                        {
+                            Console.WriteLine("Error with message order ");
```
Comment (gmltA) [inline Lab1/Program.cs:104]: So maybe fix it now?

**Review by gmltA — CHANGES_REQUESTED** [review]
I don't see how you recover message history upon reconnection.

File: Lab1/Program.cs (hunk 1 above)
Comment (maksimkha) [reply Lab1/Program.cs]: yeah, my fault

File: Lab1/Program.cs (hunk 2 above)
Comment (maksimkha) [reply Lab1/Program.cs]: added, now the initial idea is in sending "error message" to sender, when problem occurred. and when listener recieves this error message, he resends needed id

**Review by gmltA — APPROVED** [review]

**Pull request merged**